src/backend/access/external/pxfuriparser.c

```diff
@@ -719,7 +727,7 @@ GPHDUri_parse_fragment(char* fragment, List* fragments)
 		has_user_data = true;
 		*value_end = '\0';
 	}
-	fragment_data->fragment_md = pstrdup(value_start);
+	fragment_data->dfs_address = pstrdup(value_start);
```

Comment:
please make sure to pfree it in GPHDUri_free_fragments.
I also noticed that we don't free fragment_md and user_data.